### PR TITLE
[1LP][RFR] Fix test_start_pages & test_search_bar

### PR DESCRIPTION
**Pull request by gshefer**
{{ pytest: cfme/tests/containers/test_search_bar.py cfme/tests/containers/test_start_page.py --use-provider cm-env2 }}
ContainerObjectAllBaseView implementation was wrong and is_displayed didn't work properly in the container objects.

### CODE REVIEW

File: cfme/containers/image.py [hunk 1]
```diff
@@ -2,10 +2,13 @@
 from functools import partial
 import random
 import itertools
-from cached_property import cached_property
 
+from cached_property import cached_property
```
Comment (izapolsk) [inline cfme/containers/image.py]: please move cached_property back. it is system lib.

File: cfme/containers/image.py [hunk 2]
```diff
@@ -129,7 +131,7 @@ def compliant(self):
 
 
 class ImageAllView(ContainerObjectAllBaseView):
-    TITLE_TEXT = 'Container Images'
+    summary = Text('//h1[normalize-space(.) = "Container Images"]')
```
Comment (izapolsk) [inline cfme/containers/image.py]: looks a bit weird. I assume there should be easier xpath w/o text comparison. This text comparison would look better in is_displayed property.

File: cfme/containers/image_registry.py [hunk 3]
```diff
@@ -56,7 +57,7 @@ def get_random_instances(cls, provider, count=1, appliance=None):
 
 
 class ImageRegistryAllView(ContainerObjectAllBaseView):
-    TITLE_TEXT = 'Image Registries'
+    summary = Text('//h1[normalize-space(.) = "Image Registries"]')
```
Comment (izapolsk) [inline cfme/containers/image_registry.py]: same here

File: cfme/containers/provider/__init__.py [hunk 4]
```diff
@@ -440,24 +439,23 @@ def step(self):
...
-        if not hasattr(self, 'TITLE_TEXT'):
-            raise Exception('TITLE_TEXT is not defined for destination {} ("All").'
-                            .format(self.__class__.__name__))
-        return Text('//h1[normalize-space(.) = {}]'.format(quote(self.TITLE_TEXT)))
+        return self.summary
 
     @property
     def is_displayed(self):
-        return self.title.is_displayed
+        return self.summary.is_displayed
```
Comment (izapolsk) [inline cfme/containers/provider/__init__.py]: @gshefer, I think you need to define one widget summary = Text('//h1[blabla]') and create is_displayed for each view. 
Where you can compare summary.text == 'Some View Title'

**Comment by dajoRH** [comment]
Lint report for commit 5e7880dc3dbe65f793e3d527d56d1bace541ac3c:

`cfme/containers/image.py`:
- :red_circle: Line 10:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/image_registry.py`:
- :red_circle: Line 8:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/node.py`:
- :red_circle: Line 9:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/pod.py`:
- :red_circle: Line 8:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/project.py`:
- :red_circle: Line 7:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/replicator.py`:
- :red_circle: Line 8:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/route.py`:
- :red_circle: Line 8:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/service.py`:
- :red_circle: Line 8:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/template.py`:
- :red_circle: Line 9:1: **F401** *'widgetastic.widget.Text' imported but unused*

`cfme/containers/volume.py`:
- :red_circle: Line 9:1: **F401** *'widgetastic.widget.Text' imported but unused*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/containers/provider/__init__.py [hunk 5]
```diff
@@ -440,24 +439,23 @@ def step(self):
...
-        if not hasattr(self, 'TITLE_TEXT'):
-            raise Exception('TITLE_TEXT is not defined for destination {} ("All").'
-                            .format(self.__class__.__name__))
-        return Text('//h1[normalize-space(.) = {}]'.format(quote(self.TITLE_TEXT)))
+        return self.summary
 
     @property
     def is_displayed(self):
-        return self.title.is_displayed
+        return self.summary.text == self.TITLE_TEXT
```
Comment (psav) [inline cfme/containers/provider/__init__.py]: **REQUIRED:** Is ```self.title``` still needed now as all it does is return summary. Wouldn't it be better to return ```self.summary.text``` and then you can just direct compare ```self.title == self.TITLE_TEXT```

Comment (gshefer) [reply cfme/containers/provider/__init__.py]: Well, I just thought to keep this as it is for backward compatibility, but now I see that it's not really in use anywhere so maybe worth to just delete `def title` (?).

Comment (psav) [reply cfme/containers/provider/__init__.py]: Yeh if it's not used, lets delete it, particularly in this case as I think is is a little confusingly named.

**Pull request merged**